libs/dal/src/repositories/execution-details/execution-details.repository.ts

```diff
@@ -44,4 +44,26 @@ export class ExecutionDetailsRepository extends BaseRepository<
       _notificationId: notificationId,
     });
   }
+
+  /**
+   * Activity feed might need to retrieve all the executions of a notification by transactionId.
+   */
+  public async findAllNotificationExecutionsByTransactionId(
+    transactionId: string,
+    environmentId: string,
+    skip = 0,
+    limit = 10
+  ) {
+    const requestQuery = {
+      _environmentId: environmentId,
+      transactionId: transactionId,
```

Comment:
transactionId is not indexed on our end, therefore we will need to add it so we won't scan all the collection, we are using Compound Indexes so we should add it to the 
`executionDetailsSchema.index({
  _environmentId: 1,
});`
as a second index or create a new one for it, I am not entirely sure yet what will be the most optimized one.